src/async_client.rs

```diff
@@ -93,36 +93,43 @@ impl Client {
     /// Note: Since the reverse dependency endpoint requires pagination, this
     /// will result in multiple requests if the crate has more than 100 reverse
     /// dependencies.
-    pub fn crate_reverse_dependencies(
-        &self,
-        name: &str,
-    ) -> impl Future<Item = Vec<Dependency>, Error = Error> {
-        fn fetch_page(
-            c: Client,
-            name: String,
-            mut deps: Vec<Dependency>,
-            page: u64,
-        ) -> impl Future<Item = Vec<Dependency>, Error = Error> + Send {
-            let url = c
-                .base_url
-                .join(&format!(
-                    "crates/{}/reverse_dependencies?per_page=100&page={}",
-                    name, page
-                ))
-                .unwrap();
-            c.get::<Dependencies>(&url).and_then(
-                move |data| -> Box<Future<Item = Vec<Dependency>, Error = Error> + Send> {
-                    if !data.dependencies.is_empty() {
-                        deps.extend(data.dependencies);
-                        Box::new(fetch_page(c, name, deps, page + 1))
-                    } else {
-                        Box::new(::futures::future::ok(deps))
+    pub fn crate_reverse_dependencies(&self, name: &str)
+        -> impl Future<Item = ReverseDependencies, Error = Error> {
+
+        fn fetch_page(c: Client, name: String, mut tidy_rdeps: ReverseDependencies, page: u64)
+            -> impl Future<Item = ReverseDependencies, Error = Error> + Send {
+
+            let url = c.base_url.join(&format!(
+                "crates/{0}/reverse_dependencies?per_page=100&page={1}", name, page
+            )).unwrap();
+
+            c.get::<ReverseDependenciesAsReceived>(&url).and_then(move |rdeps|
+                -> Box<Future<Item = ReverseDependencies, Error = Error> + Send> {
```

Comment:
nit: this should be `dyn Future<...>`

Reply:
Ok. I don't think I'm really understanding this, but I just added the `dyn` keyword.